fiftyone/core/map/mapper.py

```diff
@@ -108,33 +122,51 @@ def map_samples(
             for sample in self._sample_collection.iter_samples(
                 progress=progress, autosave=save
             ):
-                result = map_fcn(sample)
-                yield sample.id, result
+                try:
+                    result = map_fcn(sample)
+                except Exception as err:
+                    if not skip_failures:
+                        raise err
+
+                    logger.warning(
+                        "Sample failure: %s\nError: %s\n", sample.id, err
+                    )
+                else:
+                    # Only not yield samples that have a valid result
```

Comment:
Only yield valid samples?

Reply:
Samples that were mapped successfully (did not raise an exception).